test/mlir/onnx/onnx_lowering.mlir

```diff
@@ -2132,3 +2132,17 @@ func @cast_lowering_f64f32_10(%arg0: tensor<10xf64>) -> tensor<*xf32> {
   // CHECK: affine.store [[FPTRUNC]], [[RES]][%arg1] : memref<10xf32>
   // CHECK: return [[RES]] : memref<10xf32>
 }
+
+// -----
+
+func @test_sizeof(%arg0: tensor<2x2xf32>) -> tensor<i64> {
+  %1 = "onnx.Size"(%arg0) : (tensor<2x2xf32>) -> tensor<i64>
+  "std.return"(%1) : (tensor<i64>) -> ()
+
+  // CHECK-LABEL: test_sizeof
+  // CHECK-NEXT: [[RES:%.+]] = alloc() : memref<i64>
+  // CHECK: [[SIZE:%.+]] = constant 4 : i64
+  // CHECK: affine.store [[SIZE]], [[RES]][] : memref<i64>
+  // CHECK: return [[RES]] : memref<i64>
+
```

Comment:
Please add a test for unknown dimensions? e.g. `tensor<?x?xf32>`. Thanks!

Reply:
@tungld  Added, please check.

Reply:
Thank you! 

I think we can make the emitted code better by computing the constant part at the compilation time. For example, with tensor<2x?x2xf32>, we can compute 2x2 at the compilation time, emit a constant of value 4, and emit a MulIOp for multiplication of the first dim and the constant. Something like this:
```mlir
  // Accumulate static dimensions first.
  int64_t staticNumElement = 1;
  bool allStaticDimensions = true;
  for (unsigned i = 0; i < dataShape.size(); i++) {
    if (dataShape[i] != -1)
      staticNumElement *= dataShape[i];
    else
      allStaticDimensions = false;
  }
  // Accumulate the remaining dimensions that are unknown.
  Value noElements =
      emitConstantOp(rewriter, loc, memRefType.getElementType(), staticNumElement);
  if (!allStaticDimensions) {
    for (unsigned i = 0; i < dataShape.size(); i++) {
      if (dataShape[i] == -1) {
         Value index = rewriter.create<DimOp>(loc, data, i);
         Value dim = rewriter.create<IndexCastOp>(
             loc, index, memRefType.getElementType());
         noElements = rewriter.create<MulIOp>(loc, noElements, dim);
      }
    }
  }
```

Reply:
@tungld Thanks, this looks good to me.

Reply:
Hi @tungld  Could you review further? If there is no changes could you approve?

Reply:
LGTM! Thanks!